### PR TITLE
Three New Lessons - Device Inventory, Device Backup, Template Generation

### PR DESCRIPTION
**Pull request by jweidley**
This PR introduces three new lessons to the NRE Labs curriculum:

- **Lesson 33** - Quick and Easy Device Inventory
- **Lesson 34** - Automated Device Configuration Backup
- **Lesson 35** - Device Specific Template Generation

All three lessons are contained in this pull request.

### CODE REVIEW

**Comment by jweidley** [comment]
Added stage4 to lesson 35 which covers generating a template and pushing the configuration to the vqfx switches.

**Comment by Mierdin** [comment]
@jweidley Can you provide a description for what's being contributed here? See other PRs for examples, just gives readers a sense as to what's being contributed without having to dive into the huge diff.

Also please provide a CHANGELOG.md update. See that file for examples of how this is done in the past. Since you're contributing three lessons in one PR, please still provide three separate lines, but all three will link to this PR.

File: lessons/lesson-34/stage2/guide.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+With this method new devices can be easily added without editing in the script.
+
+Next we will start the Python interactive shell, load the YAML module to process the `devices.yml` file, load the PyEz module so we can communicate with our Junos devices and load the `lxml` module to format the XML data returned from the Junos devices.
+<pre>
+python
+import yaml
+from jnpr.junos import Device
+from lxml import etree
+
```
Comment (Mierdin) [inline lessons/lesson-34/stage2/guide.md]:
```suggestion
```

File: lessons/lesson-34/stage2/guide.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+</pre>
+<button type="button" class="btn btn-primary btn-sm" onclick="runSnippetInTab('linux1', 1)">Run this snippet</button> 
+
+Next we have to read in the YAML file with the device hostnames. First, open the file as readonly and then use the YAML module to put it into a Python list, like so:
+<pre>
+deviceFile = open('devices.yml', 'r')
+deviceList = yaml.full_load(deviceFile)
+
```
Comment (Mierdin) [inline lessons/lesson-34/stage2/guide.md]:
```suggestion
```

File: lessons/lesson-34/stage2/guide.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+</pre>
+<button type="button" class="btn btn-primary btn-sm" onclick="runSnippetInTab('linux1', 2)">Run this snippet</button> 
+
+To ensure the `devices.yml` file was processed correctly we can print the `deviceList` variable.
+<pre>
+print deviceList
+
```
Comment (Mierdin) [inline lessons/lesson-34/stage2/guide.md]:
```suggestion
```

File: lessons/lesson-34/stage2/guide.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	outfile = open(device + "-backup.txt","w")
+</pre>
+<button type="button" class="btn btn-primary btn-sm" onclick="runSnippetInTab('linux1', 5)">Run this snippet</button>
+
+Then we will use the `rpc.get_config` function to pull the device configuration and store it in a variable called `config`. Next we write the configuration to the local file and finally `close` the local file.
+<pre>
+	config = dev.rpc.get_config(options={'format':'text'})
+	outfile.write(etree.tostring(config))
+	outfile.close()
+
```
Comment (Mierdin) [inline lessons/lesson-34/stage2/guide.md]:
```suggestion
```

File: lessons/lesson-34/syringe.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
+---
+lessonName: Automated Device Configuration Backup
+lessonId: 34
+category: workflows
+lessonDiagram: 
+tier: prod
+prereqs:
+  - 24  # PyEz
```
Comment (Mierdin) [inline lessons/lesson-34/syringe.yaml]:
```suggestion
  - 24  # PyEz
  - 14  # YAML
```

File: lessons/lesson-33/stage1/guide.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
+# Automated Device Inventory
```
Comment (Mierdin) [inline lessons/lesson-33/stage1/guide.md]: I won't bother repeating this comment for every single lab guide, but please give yourself/yourselves credit for building this lesson. In the near future we'll have a feature where you can define this in the lesson definition, but for now, please use https://github.com/nre-learning/nrelabs-curriculum/blob/master/lessons/lesson-26/stage1/guide.md as an example for how this should be done. Please do this on top of each lab guide for each stage, for each lesson.

**Pull request merged**